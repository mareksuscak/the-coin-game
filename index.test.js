const fs = require('fs');
const canWin = require('./index');

// Read the input file and convert each non-empty line into a Number
const Ns = fs.readFileSync('coins.in')
  .toString()
  .split(/\r?\n/)
  .filter((line) => line !== '')
  .map(Number)

test('computes the result correctly for all Ns', () => {
  expect(Ns.map(canWin)).toMatchSnapshot();
});
