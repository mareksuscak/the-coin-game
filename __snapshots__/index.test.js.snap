// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computes the result correctly 1`] = `
Array [
  true,
  false,
  true,
  true,
  false,
  true,
  false,
  true,
  true,
  false,
  true,
  false,
  true,
  true,
  false,
  true,
  false,
  true,
  true,
  false,
  false,
  true,
  false,
  false,
  false,
  true,
  true,
  false,
  true,
  true,
  true,
  true,
  true,
  false,
  true,
  false,
  false,
  true,
  true,
  true,
  false,
  false,
  true,
  false,
  false,
  false,
  false,
  false,
  true,
  false,
  false,
  true,
  false,
  false,
  true,
  false,
  true,
  false,
  true,
  true,
  false,
  true,
  true,
  false,
  true,
  false,
  false,
  false,
  false,
  false,
  true,
  true,
  true,
  false,
  false,
  true,
  false,
  false,
  true,
  true,
  false,
  true,
  true,
  true,
  false,
  false,
  false,
  false,
  true,
  true,
  false,
  true,
  true,
  false,
  false,
  false,
  true,
  true,
  false,
  true,
  false,
  false,
  true,
  false,
  true,
  false,
  true,
  false,
  false,
  true,
  true,
  false,
  false,
  true,
  true,
  true,
  false,
  false,
  true,
  false,
]
`;
